ports/nrf/mpconfigport.h

```diff
@@ -312,6 +312,7 @@ extern const struct _mp_obj_module_t ble_module;
     const char *readline_hist[8]; \
     mp_obj_t pin_class_mapper; \
     mp_obj_t pin_class_map_dict; \
+    struct _machine_pin_irq_handler_t * machine_pin_irq_handlers; \
```

Comment:
You could move the entire `irq_handlers` over to here I think, just like `readline_hist` for example.

Reply:
The reason why i put it in pin.c is that i had a very hard time resolving the NUMBER_OF_PINS from the include hierarchy of nrfx.h inside this header. i'll give it a second try.

Reply:
This is actually required: the heap allocated pointers must be directly stored in here otherwise the GC won't pick up the root pointers. If it is done via an indirect (.bss) pointer, the GC will consider the pointer to be outside the heap and won't scan it any further.